packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -616,6 +646,17 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
     }
   }, [connectionState, pushToast])
 
+  /**
+   * If the document was deleted successfully, close the pane.
+   */
+  useEffect(() => {
+    if (!documentOperationEvent) return
+    if (documentOperationEvent.type === 'success' && documentOperationEvent.op === 'delete') {
+      // Wait until next tick to allow deletion toasts to appear first
```

Comment:
Moved it back to the provider because it felt like `DocumentOperationResults` was only concerned with toasts. I believe using a `setTimeout()` does avoid the race condition, so we can do either.